src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -142,33 +144,98 @@ public static IEnumerable<LowLevelClientMethod> BuildMethods(RestClientBuilder b
             }
         }
 
-        private IEnumerable<ConstructorSignature> BuildPublicConstructors()
+
+        private (ConstructorSignature[] PrimaryConstructors, ConstructorSignature[] SecondaryConstructors) BuildPublicConstructors(IReadOnlyList<Parameter> orderedParameters)
         {
-            if (!_hasPublicConstructors)
+            if (!IsSubClient)
+            {
+                var requiredParameters = RestClientBuilder.GetRequiredParameters(orderedParameters).ToArray();
+                var optionalParameters = RestClientBuilder.GetOptionalParameters(orderedParameters).Append(CreateOptionsParameter()).ToArray();
+
+                return (
+                    BuildPrimaryConstructors(requiredParameters, optionalParameters).ToArray(),
+                    BuildSecondaryConstructors(requiredParameters, optionalParameters).ToArray()
+                );
+            }
+            else
             {
-                yield break;
+                return (Array.Empty<ConstructorSignature>(), new[] { CreateMockingConstructor() });
             }
+        }
 
-            var clientOptionsType = ClientOptions.Type.WithNullable(true);
-            var clientOptionsParameter = new Parameter("options", "The options for configuring the client.", clientOptionsType, Constant.Default(clientOptionsType), ValidationType.None, Constant.NewInstanceOf(clientOptionsType).GetConstantFormattable());
+        private IEnumerable<ConstructorSignature> BuildPrimaryConstructors(IReadOnlyList<Parameter> requiredParameters, IReadOnlyList<Parameter> optionalParameters)
+        {
+            var optionalToRequired = optionalParameters.Select(parameter =>
+                ClientOptions.Type.EqualsIgnoreNullable(parameter.Type) ?
+                parameter with { DefaultValue = null, Validation = ValidationType.None } :
+                Parameter.FromOptionalToRequired(parameter)).ToArray();
```

Comment:
I'd rather make `Parameter.GetValidation` public and inline `FromOptionalToRequired` , cause this method has meaning only in this case, e.g.:
```cs
var optionalToRequired = optionalParameters
    .Select(parameter => ClientOptions.Type.EqualsIgnoreNullable(parameter.Type)
        ? parameter with { DefaultValue = null, Validation = ValidationType.None }
        : parameter with
        {
            DefaultValue = null,
            Validation = parameter.Initializer != null ? Parameter.GetValidation(parameter.Type, parameter.RequestLocation, parameter.SkipUrlEncoding) : parameter.Validation,
            Initializer = null
        }
    )
    .ToArray();
```